cli: include jobs and more in debug zip

tbg opened the pull request:
Also clean up a bit. We were collecting some information only from
the local node and not from the other nodes.

Release note (general change): `cockroach debug zip` now contains more
information.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/35880)
<!-- Reviewable:end -->

bdarnell reviewed (approved):
> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/35880)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

tbg commented:
Ignoring the rampant sql flake

bors r=bdarnell

tbg committed:
cli: include jobs and more in debug zip

Also clean up a bit. We were collecting some information only from
the local node and not from the other nodes.

Release note (general change): `cockroach debug zip` now contains more
information.

tbg commented:
bors r=bdarnell
